src/Fhir.Anonymizer.Shared.Core/Processors/SubstituteProcessor.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Fhir.Anonymizer.Core.Extensions;
+using Fhir.Anonymizer.Core.Models;
+using Fhir.Anonymizer.Core.Utility;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Hl7.FhirPath.Sprache;
+
+namespace Fhir.Anonymizer.Core.Processors
+{
+    public class SubstituteProcessor : IAnonymizerProcessor
+    {
+        private readonly FhirJsonParser _parser = new FhirJsonParser();
+
+        public ProcessResult Process(ElementNode node, ProcessSetting setting = null)
+        {
+            if (setting == null)
+            {
+                setting = new ProcessSetting
+                {
+                    ReplaceWith = string.Empty,
+                    IsPrimitiveReplacement = true,
+                    VisitedNodes = new HashSet<ElementNode>()
+                };
+            }
+            else if (setting.ReplaceWith == null)
+            {
+                setting.ReplaceWith = string.Empty;
+            }
+            else if (setting.VisitedNodes == null)
+            {
+                setting.VisitedNodes = new HashSet<ElementNode>();
+            }
+
+            ElementNode replacementNode;
+            // Get replacementNode for substitution  
+            if (setting.IsPrimitiveReplacement)
+            {
+                replacementNode = ElementNode.FromElement(ElementNode.ForPrimitive(setting.ReplaceWith));
+            }
+            else
+            {
+                var modelAssembly = typeof(Patient).Assembly;
+                var replacementNodeType = modelAssembly
+                    .GetTypes()
+                    .Where(type => string.Equals(type.Name, node.InstanceType, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
We should put this logic to general class. It should cache Resource Types and take resource type and json text input.

Reply:
Thanks. Changed to use the ModelInfo.GetTypeForFhirType API from Hl7 library